api/samples/memval_simple.c

```diff
@@ -219,8 +219,7 @@ instrument_mem(void *drcontext, instrlist_t *ilist, instr_t *where, opnd_t ref)
     size = (ushort)drutil_opnd_mem_size_in_bytes(ref, where);
     drx_buf_insert_buf_store(drcontext, trace_buffer, ilist, where, reg_ptr, reg_tmp,
                              OPND_CREATE_INT16(size), OPSZ_2, offsetof(mem_ref_t, size));
-    drx_buf_insert_update_buf_ptr(drcontext, trace_buffer, ilist, where, reg_ptr,
-                                  DR_REG_NULL, sizeof(mem_ref_t));
+    /* postpone updating trace_buffer ptr to post-write, in case the write segfaults */
```

Comment:
I'm not sure I understand: is it the *app* write faulting, or the *trace buffer write* faulting?

Reply:
The app write is faulting. There are some writes[1] in `drx-scattergather.c` that intentionally `SIGSEGV` to verify whether the scratch mask register is restored upon a fault.

[1]: https://github.com/DynamoRIO/dynamorio/blob/cf9d6a95262015581a5184e75ce599cc66ac4df4/suite/tests/client-interface/drx-scattergather.c#L569

Reply:
OK, some of these buffers use faults to handle filling up, so just making sure I understand.  The comment does imply it's the app, but maybe making it even clearer could help.

Reply:
Done.

Reply:
Will push with later commits.